Source/WebSocket/Websocketpp/websocketpp_websocket.cpp

```diff
@@ -390,6 +393,22 @@ struct wspp_websocket_impl : public hc_websocket_impl, public std::enable_shared
                 auto context = http_allocate_shared<client_context>(client);
                 
                 m_websocketThread = std::thread([context](){
+#if HC_PLATFORM == HC_PLATFORM_ANDROID
+                    auto httpSingleton = xbox::httpclient::get_http_singleton(true);
+                    AndroidPlatformContext* platformContext = reinterpret_cast<AndroidPlatformContext*>(httpSingleton->m_platformContext.get());
```

Comment:
since client.run() is blocking, this auto singleton will keep our singleton alive too long..

Wrap these two lines inside { } so the singleton goes out of scope quickly